parasol/solr/admin.py

```diff
@@ -0,0 +1,32 @@
+from urllib.parse import urljoin
+
+from attrdict import AttrDict
+
+class CoreAdmin:
+
+    def __init__(self, client):
+        self.client = client
```

Comment:
Passing in the client here still seems awkward to me. Maybe just the session should be passed in? (and could be optional)

What else do the other API classes use from the main client class, just `make_request` ?

Reply:
`make_request` and the ability to build base urls, but that seems like something that can be abstracted to a base class, which I've done and seems pretty clean so far.